Update: Improve a guide about how to use executeTx and ExecParams (#772)
scale-it/algo-builder: docs/guide/execute-transaction.md

# Execute Transaction

> `executeTx` is a high level method of Deployer which can be used to perform transactions on Algorand Network. It supports every transaction (atomic or single) which is possible in network. Ex: Deploy ASA/App, Opt-In, Transfers, Delete, Destroy etc. `executeTx` takes `ExecParams[]` as parameter. If the length of `ExecParams` array is greater than one, it will be considered as `atomic transaction` otherwise `single transaction`.

Reviewer: it also takes `SignedTransaction`

Author: Ok. Done.

> `executeTx` is a high level method of Deployer which can be used to perform transactions on Algorand Network. It supports every transaction (atomic or single) which is possible in network. Ex: Deploy ASA/App, Opt-In, Transfers, Delete, Destroy etc. `executeTx` takes `ExecParams[]` or `SignedTransaction[]` as parameter. If the length of `ExecParams` array is greater than one, it will be considered as `atomic transaction` otherwise `single transaction`.

Reviewer: Lets add a section where we shortly describe `ExecParams` and `SignedTransaction`. The differences and why would one prefer to use one over another

Author: I have created a section for ExecParams and we already have a section for SignedTransaction. I have also highlighted the difference between them.

Depending on the transaction `type`, other attributes will specify a signer of the transaction. For example, for `TransactionType.TransferAlgo` the `fromAccount` must be a full account (with secret key) and will sign the transaction.

> `deployer.executeTx` returns list of `TxnReceipt`, which extends `ConfirmedTxInfo`.

Reviewer: lets also add example with `signedTransaction`

Depending on the transaction `type`, other attributes will specify a signer of the transaction. For example, for `TransactionType.TransferAlgo` the `fromAccount` must be a full account (with secret key) and will sign the transaction.

`deployer.executeTx` returns list of `TxnReceipt`, which extends `ConfirmedTxInfo`.

```js
const receipts = deployer.executeTx([txn0, txn1]);
console.log("txn0 information: ", receipts[0]);
console.log("txn1 information: ", receipts[2]);
```

> Below examples demonstrates, how to perform application call using `ExecParam` and `WebMode`.

Reviewer 2: ```suggestion
Below we demonstrate how to perform application call using `ExecParam` and `WebMode`:
```

```js
appCall = () => {
		const webMode: WebMode = new WebMode(AlgoSigner, networkType);
		const tx: types.ExecParams[] = [
			{
				type: types.TransactionType.CallApp,
				sign: types.SignType.SecretKey,
				fromAccount: {
					addr: addr,
					sk: new Uint8Array(0),
				},
				appID: 100,
				payFlags: {},
			},
		];
		webMode.executeTx(tx);
	};
```

> We can also use `SignedTransaction` and `executeTx` in runtime:

Reviewer: we could add link to an example from Algorand docs with the signedTransaction. How it is created using `algosdk`. Or just add it to our example

Author: We already have signed transaction example at the end of this doc.

> Below examples demonstrates, how to use `SignedTransaction` and `executeTx` in runtime.

Reviewer 2: ```suggestion
We can also use `SignedTransaction` and `executeTx` in runtime:
```
